spec/book_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'date'
+require_relative '../classes/book'
+
+describe Book do
+  before(:each) do
+    @book_item = Book.new('Harry Potter', 'JK Rowlings', 'good', '2018-05-12')
+  end
+  context 'Book class' do
+    it 'A newly created book has a title of type String' do
+      expect(@book_item.name).not_to be_nil
+      expect(@book_item.name).to be_a(String)
+    end
+
+    it ' detect the book title' do
+      expect(@book_item.name).to eq 'Harry Potter'
+    end
+
+    it ' detect the book publisher' do
+      expect(@book_item.publisher).to eq 'JK Rowlings'
+    end
+
+    it ' detect the book cover_state' do
+      expect(@book_item.cover_state).to eq 'good'
+    end
+
+    it 'detect the book publish_date' do
+      expect(Date.parse(@book_item.publish_date)).to eq Date.parse('2018-05-12')
+    end
+
+    it 'detect the book archive state' do
+      expect(@book_item.archived).to be false
+    end
+
```

Comment:
- [ ] As required for the project that all implemented methods for a `Class` should have unit tests, I suggest you also add tests to the `#can_use_service?` from the `Book` class as you did for other methods in different classes

Reply:
@tresorsawasawa can you explain this since there is not method called can_use_service?

Reply:
Hi @rbhatt1999 

Sorry, this is a typo error
The expected method is `can_be_archived?`

Reply:
@tresorsawasawa `can_be_archived?` is private method we can't call it please explain how can we run a test on it